src/main/java/ev3dev/sensors/GenericMode.java

```diff
@@ -0,0 +1,80 @@
+package ev3dev.sensors;
+
+import ev3dev.utils.Sysfs;
+import lejos.hardware.sensor.SensorMode;
+
+import java.io.File;
+
+public class GenericMode implements SensorMode {
+
+    private final String VALUE = "value";
+    private final String VALUE0 = "value0";
+
+    private final File pathDevice;
+    private final int sampleSize;
+    private final String name;
+    private final int type;
+    private float minRange;
+    private float maxRange;
+    private float correctionFactor;
+
+    public GenericMode (
+            final File pathDevice,
+            final int sampleSize,
+            final String name,
+            final int type) {
+        this.pathDevice = pathDevice;
+        this.sampleSize = sampleSize;
+        this.name = name;
+        this.type = type;
+    }
+
+    public GenericMode(
+            final File pathDevice,
+            final int sampleSize,
+            final String name,
+            final int type,
+            final float minRange,
+            final float maxRange,
+            final float correctionFactor) {
+        this(pathDevice, sampleSize, name, type);
+        this.minRange = minRange;
+        this.maxRange = maxRange;
+        this.correctionFactor = correctionFactor;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public int sampleSize() {
+        return sampleSize;
+    }
+
+    @Override
+    public void fetchSample(float[] sample, int offset) {
+
```

Comment:
I think there lies a possible bug. If you create two different modes after one another, sensor-getter of the second one will change the underlying sensor mode and affect the first GenericMode. Mode switching code should live here (I remember seeing it in leJOS implementation of SampleProvider).

Reply:
But what I've seen was most probably an instance of inner class we now try to avoid. I see now that the mode switching logic lives in BaseSensor class. Maybe the switching logic could be factored out so that it is possible to do it here.

Reply:
Hmmm, actually, maybe we can just include a BaseSensor instance in GenericMode. Then, copying the `switchMode` lines would be enough.

Reply:
Indeed, possibility to use sensor modes concurrrently was the motivation for this machinery in leJOS. https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/121#issuecomment-205161182

Reply:
Hmmm, you could add a specialized modeswitch function to BaseSensor which would take just a sensor mode as an argument. The rest of the parameters could be stored in the BaseSensor itself, IIUC.